### DISCUSSION TITLE
GetLifeSpan Polling

### DISCUSSION THREAD

**Discussion by edenhaus** [And3rsL/Deebotozmo#35] [Ideas]
With the new refactoring I subscribed for updates on MQTT. Unfortunately this is not working for LifeSpan Events, so we need still to poll this information.

Currently we poll it each minute, which are 1440 request in 24h. This is a lot if you think that it will change only during cleaning.
To reduce it more I have come up with some ideas:

- Subscribe to the onSleep Event and poll only the information, when the robot is not sleeping
- Poll only during cleaning (maybe we miss it, if someone is reseting it)

@And3rsL What do you think or maybe have you a better idea?

**Comment by Edo78** [comment]
The lifespan doesn't change so often to need a 1 minute interval.
If there's not a specific reason to poll each minute I'd suggest as a first approach to extend the polling interval to 24 hours.

The teorical best approach should be to poll only during cleaning (or just one when cleaning ends) but I'm not sure how reliable are those events.

I haven't already reach a 0% lifespan on any components so I'm not 100% sure but I'm pretty confident that the robot can still works and it shouldn't explode ;)

**Reply by edenhaus** [reply]
In the past we polled this information each minute and therefore I left it unchanged for backwards compatibility. For me it's okay to poll this information only once a day, but I don't if others need this information more accurate. I would suggest to use one of my suggestions as we get the information at the same level during cleaning

When you reach 0% nothing will happen, only ecovacs gives you the advice to change the specific component. Ex. the filter

**Comment by edenhaus** [comment]
Implemented now, that we are polling the lifespan events only during cleaning
